--- original/data/lib/smartcard/pcsc/card.rb	1970-01-01 01:00:00.000000000 +0100
+++ smartcard-0.5.1.gem_FILES/data/lib/smartcard/pcsc/card.rb	2011-02-02 22:21:59.737730001 +0100
@@ -36,8 +36,8 @@ class Card
     
     @context = context
     @sharing_mode = sharing_mode
-    @_handle = handle_ptr[:value]    
-    set_protocol FFILib::Protocol[protocol_ptr[:value]]
+    @_handle = handle_ptr[:value]
+    set_protocol FFILib::Protocol.to_h.invert[protocol_ptr[:value]]
   end
   
   # Updates internal buffers to reflect a change in the communication protocol.
@@ -88,7 +88,7 @@ class Card
     raise Smartcard::PCSC::Exception, status unless status == :success
   
     @sharing_mode = sharing_mode
-    set_protocol FFILib::Protocol[protocol_ptr[:value]]
+    set_protocol FFILib::Protocol.to_h.invert[protocol_ptr[:value]]
   end
   
   # Disconnects from the smart-card.
@@ -268,7 +268,7 @@ class Card
         end
         
         { :readers => Context.decode_multi_string(readers_ptr),
-          :protocol => FFILib::Protocol[protocol_ptr[:value]],
+          :protocol => FFILib::Protocol.to_h.invert[protocol_ptr[:value]],
           :atr => atr_ptr.get_bytes(0, atr_length_ptr[:value]),
           :state => state }
       ensure
